src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -342,67 +357,96 @@ private void validateRowIdsForRoot(String studyDatasetName,
         }
     }
 
-    // Recursive walk of the asset relationships. Note that we only follow what is connected.
-    // If there are relationships in the asset that are not connected to the root, they will
-    // simply be ignored. See the related comment in study validator.
+    /**
+     * Recursive walk of the relationships. Note that we only follow what is connected.
+     * If there are relationships in the asset that are not connected to the root, they will
+     * simply be ignored. See the related comment in study validator.
+     *
+     * We operate on a pdoa-specific list of the asset relationships so that we can
+     * bookkeep which ones we have visited. Since we need to walk relationships in both
+     * the from->to and to->from direction, we have to avoid re-walking a traversed relationship
+     * or we infinite loop. Trust me, I know... :)
+     *
+     * TODO: REVIEWERS: should this code detect circular references?
```

Comment:
It would be nice to have in this PR, but I don't think it's a quick couple of lines of code to write. If you feel like it's out of the scope of this work can you add something in the backlog to track?